Utt_Christelle_Homework_Software_Engineering_1.py

```diff
@@ -0,0 +1,138 @@
+"""
+TASK 1
+Write a class to represent a Cash Register.
+You class should keep the state of price total and purchased items
+Below is a starter code:
+------------------------
+1. you can add new variables and function if you want to
+2. you will NEED to add accepted method parameters where required.
+For example, method add_item probably accepts some kind of an item?..
+3. You will need to write some examples of how your code works.
+"""
+#PS! Printing is only for logging purposes.
+
+class CashRegister:
+
+    def __init__(self):
+        self.total_items = {} # {'item': price}
+        self.total_price = 0
+        self.discount = 0
+
+    def add_item(self, item, price):
+        if type(price) != int and type(price) != float:
+            return
+        price -= price * self.discount
+        self.total_items[item] = price
+        self.total_price += price
+        print(f"You have added {item} into your basket.")
+
+    def remove_item(self, item):
+        self.total_items.pop(item)
+        print(f"{item.title()} is removed from your basket.")
+
+    def apply_discount(self, discount):
+        if 0 < discount < 1:
+            self.discount = discount
+            print(f"Discount {discount*100:02.0f}% is applied.")
+
+    def get_total(self):
+        print(f"Total price is {self.total_price:02.2f} euros.")
+        return self.total_price
+
+    def show_items(self):
+        print(f"In your basket, there are: {self.total_items}.")
+        return self.total_items
+
+    def reset_register(self):
+        print(f"You're basket is clear.")
+        self.total_items.clear()
```

Comment:
⚠️ reset should reset and clear the total_price and discount as well to initial value